build-support/bin/ci.sh

```diff
@@ -104,13 +105,26 @@ esac
 # We're running against a Pants clone.
 export PANTS_DEV=1
 
+# Determine interpreter for both under-the-hood and for subprocesses.
+# Order matters here. We must constrain subprocesses before running the bootstrap stage,
+# or we will encounter the _Py_Dealloc error when running `./pants.pex -V` with Python 3 under-the-hood.
+if [[ "${python_two:-false}" == "false" ]]; then
+  py_version_number="3"
+  pants_script="./pants3"
+  export PANTS_PYTHON_SETUP_INTERPRETER_CONSTRAINTS='["CPython>=3.6,<4"]'
+else
+  py_version_number="2"
+  pants_script="./pants"
+fi
+banner "Using Python ${py_version_number} to execute subprocesses."
+
 if [[ "${run_bootstrap:-true}" == "true" ]]; then
-  start_travis_section "Bootstrap" "Bootstrapping pants"
+  start_travis_section "Bootstrap" "Bootstrapping pants with Python ${py_version_number} under-the-hood."
   (
     if [[ "${run_bootstrap_clean:-false}" == "true" ]]; then
       ./build-support/python/clean.sh || die "Failed to clean before bootstrapping pants."
     fi
-    ./pants binary \
+    ${pants_script} binary \
```

Comment:
Maybe call this `bootstrap_pants_script` to make it clear the difference between the `./pants`/`./pants3` and the `./pants`/`./pants.pex` divide?